spec/spec.md

```diff
@@ -1597,6 +1597,84 @@ JSONPath                      | Description
 
 ## Appendix
 
+### Security Considerations
+JSONPath and JSON schemas are being used in this specification. Both allow the use of Regular Expressions, which have 
+numerous different (often incompatible) implementations. Some implementations allow the embedding of arbitrary code, 
+which MUST never be permitted, given the presentation definition and submission data should be treated as input 
+from untrusted sources. 
+Regular expressions can often also be crafted to be extremely expensive to compute, resulting in
+denial-of-service attacks.
+
+As such this specification recommends not using Regular Expressions at all in your implementation when using JSONPath 
+and JSON Schemas, at the potential expense of expressiveness and/or usability. If you really want to use them, make sure
+the code is properly guarded both in terms of execution time and from a security perspective, by for instance 
+executing the code dealing with the regular expressions in a sandbox. 
+
+When allowing Regular Expression Implementers need to be aware that good average performance is not sufficient as long
+as an attacker can choose to submit specially crafted Regular Expressions that can result in surprisingly high, possibly
+exponential, CPU usage or for example result in a stack overflow. Also when you do allow Regular Expressions in your
+solution, for instance in the Presentation Definition, you will risk the other side not supporting it, as they took the
+ cautious route of not allowing Regular Expressions. This specification is still suggesting to not allow Regular 
+Expressions, to not break backwards compatibility with version 2.0.0. A future new major version is highly likely to 
+not allow for usage of Regular Expressions in a normative way.
+
+#### JSONPath
+
+JSONPath is being used in this specification both in the Input Descriptors of the Presentation Definition,
+defining where certain data is to be found, and the Submission Data,
+defining how data maps from the definition to the Verifiable Presentation.
+With Regular Expressions come certain security considerations, that must be taken into account.
+Be aware that in a future major version of the spec we envision to have some normative texts in the specification
+itself disallowing certain aspects of JSONPath altogether like function extensions (ie regular expressions)
+
+Please also read the JSONPath IETF document providing more details on some of the security considerations mentioned
```

Comment:
pointing to a concrete url would be nice https://datatracker.ietf.org/api/v1/doc/document/draft-ietf-jsonpath-base/

Reply:
Yeah thanks, still need to do those. This is just a draft to solicit some feedback